src/order.rs

```diff
@@ -0,0 +1,117 @@
+use std::cmp::Ordering;
+use std::collections::BTreeSet;
+
+use async_graphql::{ComplexObject, Enum, Result, SimpleObject};
+use bson::Uuid;
+use bson::{datetime::DateTime, Bson};
+use serde::{Deserialize, Serialize};
+
+use crate::order_datatypes::OrderDirection;
+use crate::{
+    order_datatypes::CommonOrderInput, order_item::OrderItem,
+    order_item_connection::OrderItemConnection, user::User,
+};
+
+/// The Order of a user.
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone, SimpleObject)]
+#[graphql(complex)]
+pub struct Order {
+    /// Order UUID.
+    pub _id: Uuid,
+    /// User.
+    pub user: User,
+    /// Timestamp when Order was created.
+    pub created_at: DateTime,
+    /// The status of the Order.
+    pub order_status: OrderStatus,
+    /// Timestamp of Order placement. `None` until Order is placed.
+    pub placed_at: Option<DateTime>,
+    /// The rejection reason if status of the Order is `OrderStatus::Rejected`.
+    pub rejection_reason: Option<RejectionReason>,
+    pub internal_order_items: BTreeSet<OrderItem>,
```

Comment:
docs?

Reply:
This should not be public, docs would show in the GraphQL interface. Will fix.